base/lttagging.dtx

```diff
@@ -0,0 +1,789 @@
+                \int_compare:nNnTF \g_@@_table_cols_tl = 0
+                    { outer~ level }
+                    { max:~ \g_@@_table_cols_tl }
+                )
+              }
+}
+%    \end{macrocode}
+%  \end{macro}
+%
+%
+%
+%  \begin{macro}{\tbl_update_multicolumn_cell_data:n}
+%    This macro updates \cs{g_@@_col_int} and \cs{g_@@_span_tl} inside
+%    a \cs{multicolumn} and possibly calls the tagging socket
+%    \texttt{tbl/row/begin}.
+%    \begin{macrocode}
+\cs_new_protected:Npn \tbl_update_multicolumn_cell_data:n #1 {
+%    \end{macrocode}
+%    We execute socket for tagging only if this \cs{multicolumn}
+%    replaces the preamble of the first column. In that case we also have
+%    to set \cs{g_@@_col_int} to 1 because this is no longer done in the
+%    preamble for the cell either.
+%    \begin{macrocode}
+   \int_compare:nNnTF \g_@@_col_int = 0
+       {
+         \UseTaggingSocket{tbl/row/begin}
+         \int_gset:Nn \g_@@_col_int {1}
+       }
+%    \end{macrocode}
+%    If we are in a later column we use \cs{g_@@_span_tl} from the
+%    previous column to update.
+%    \begin{macrocode}
+       {
+         \int_gadd:Nn \g_@@_col_int { \g_@@_span_tl }
+       }
+%    \end{macrocode}
+%    Then we set the span value so that it can be use in the next column.
+%    \begin{macrocode}
+   \tl_gset:Nn \g_@@_span_tl {#1}
+}
+%    \end{macrocode}
+%  \end{macro}
+%
+%
+%
+%  \begin{macro}{\tbl_crcr:n}
+%    This macro is used instead of the usual \cs{crcr} at the end of a
+%    table. It is deliberately defined without protection because it
+%    may get expanded by the scanning mechanism of low-level \TeX{}
+%    after a final \cs{cr} (aka \verb=\\=) in the table. In that case
+%    it shouldn't stop the expansion and the conditional inside will
+%    be false, thus it just vanishes without doing anything. If there
+%    are missing cells (in which case we also haven't see \cs{cr} yet)
+%    the macro \cs{tbl_count_missing_cells:n} is executed and
+%    then the row is finished with a final \cs{cr}.
+%    \begin{macrocode}
+\cs_new:Npn \tbl_crcr:n #1 {  
+    \int_compare:nNnT \g_@@_col_int > 0
+        {
+          \tbl_count_missing_cells:n {#1}
+          \cr
+        }
+}
+%    \end{macrocode}
+%  \end{macro}
+%
+%
+%
+%    \begin{macrocode}
+\ExplSyntaxOff
+%<@@=>
+%    \end{macrocode}
+%
+%    This is needed for \pkg{longtable} because \cs{refstepcounter} is
+%    setting up a target when \pkg{hyperref} is loaded and we don't
+%    want that in \pkg{longtable}.
+%
+%    TODO: move to right .dtx file
+%    \begin{macrocode}
+\let\@kernel@refstepcounter\refstepcounter
+%    \end{macrocode}
+%    And no patch by hyperref (should be in longtable in the end)
+%    \begin{macrocode}
+\def\hyper@nopatch@longtable{}
+%    \end{macrocode}
+%
+%    Account for removed file (for now)
+%    \begin{macrocode}
+\ifcsname bitset_new:Nn\endcsname
+  \declare@file@substitution{l3bitset.sty}{null.tex}
+\fi
+%    \end{macrocode}
+%
```

Comment:
this could go if a current pdfmanagement and tagpdf is moved into the texmf.

Reply:
and when exactly is that?

Reply:
it could be done an any time. I prepared a branch to move them into develop a few days ago, but got hold up as the tests could not run. I can run the tests now and merge into develop and you could pull into your branch. OK?